catalog/services_state_test.go

```diff
@@ -527,6 +529,40 @@ func Test_ClusterMembershipManagement(t *testing.T) {
 	})
 }
 
+func Test_DecodeStream(t *testing.T) {
+	Convey("When managing cluster members", t, func() {
+		serv := service.Service{ID: "007", Name: "api", Hostname: "some-aws-host", Status: 1}
+		state := NewServicesState()
+		state.AddServiceEntry(serv)
+
+		httpmock.Activate()
```

Comment:
It's good that you are testing this thing end to end the way you're planning to use it, but since it only needs an `io.Reader` you should probably just [use a `bytes.Buffer`](https://golang.org/pkg/bytes/#Buffer) which will make the test smaller and faster and easier to understand.